Make imports work with `moduleResolution: "node"`

On a new Next.js app, when you do `import { remarkCodeHike, recmaCodeHike } from "codehike/mdx"` typescript complains:

```
Cannot find module 'codehike/mdx' or its corresponding type declarations.
  There are types at 'node_modules/codehike/dist/mdx.d.ts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'.ts(2307)
```

To fix it you need to change `moduleResolution` to `bundler` in the `tsconfig.json`. Would be nice to not have to do that.

--- a/.changeset/violet-insects-melt.md
+++ b/.changeset/violet-insects-melt.md
@@ -1,0 +1,5 @@
+---
+"codehike": patch
+---
+
+Compatibility with ts `moduleResolution: node`, fix #437

--- a/packages/codehike/package.json
+++ b/packages/codehike/package.json
@@ -23,19 +23,27 @@
     "type": "github",
     "url": "https://github.com/sponsors/code-hike"
   },
+  "files": [
+    "dist/**"
+  ],
   "exports": {
     "./package.json": "./package.json",
     ".": "./dist/index.js",
     "./mdx": "./dist/mdx.js",
     "./blocks": "./dist/blocks.js",
-    "./code": "./dist/code/index.js",
+    "./code": "./dist/code.js",
     "./utils/token-transitions": "./dist/utils/token-transitions.js",
     "./utils/static-fallback": "./dist/utils/static-fallback.js",
     "./utils/selection": "./dist/utils/selection.js"
   },
-  "files": [
-    "dist/**"
-  ],
+  "types": "index.d.ts",
+  "typesVersions": {
+    "*": {
+      "*": [
+        "dist/*"
+      ]
+    }
+  },
   "scripts": {
     "build": "tsc -p . ",
     "dev": "tsc -p . --watch",

--- a/packages/codehike/src/code.tsx
+++ b/packages/codehike/src/code.tsx
@@ -16,11 +16,11 @@ import type {
   CustomToken,
   CustomTokenWithAnnotation,
   InlineProps,
-} from "./types.js"
+} from "./code/types.js"
 
-import { highlight } from "./highlight.js"
-import { Pre, Inline } from "./pre.js"
-import { InnerPre, getPreRef, InnerLine, InnerToken } from "./inner.js"
+import { highlight } from "./code/highlight.js"
+import { Pre, Inline } from "./code/pre.js"
+import { InnerPre, getPreRef, InnerLine, InnerToken } from "./code/inner.js"
 
 export type {
   RawCode,

